chemprop/sklearn_predict.py

```diff
@@ -35,8 +30,8 @@ def predict_sklearn(args: SklearnPredictArgs) -> None:
     print('Computing morgan fingerprints')
     morgan_fingerprint = get_features_generator('morgan')
     for datapoint in tqdm(data, total=len(data)):
-        datapoint.set_features(morgan_fingerprint(mol=datapoint.smiles, radius=train_args.radius, num_bits=train_args.num_bits))
-
+        [datapoint.extend_features(morgan_fingerprint(mol=s, radius=train_args.radius, num_bits=train_args.num_bits))
+         for s in datapoint.smiles]
```

Comment:
I think this code will function as intended but it's not super typical to do a list comprehension without setting a variable equal to the result. Instead, it might be better to do a regular for loop as follows:

```suggestion
        for s in datapoint.smiles:
            datapoint.extend_features(morgan_fingerprint(mol=s, radius=train_args.radius, num_bits=train_args.num_bits))
```

Reply:
This also applies to a similar place in `sklearn_train.py`.